R/jira.R

```diff
@@ -8,103 +8,392 @@
+                             id = "2",
+                             description = "A new feature of the product, which has yet to be developed.",
+                             iconUrl = "https://domain.org/jira/secure/viewavatar?size=xsmall&avatarId=21141&avatarType=issuetype",
+                             name = "New Feature",
+                             subtask = FALSE,
+                             avatarId = 21141) {
+  issuetype <- list(
+    self = list(list(self_url)),
+    id = list(list(id)),
+    description = list(list(description)),
+    iconUrl = list(iconUrl),
+    name = list(name),
+    subtask = list(subtask),
+    avatarId = list(avatarId)
+  )
+
+  return(issuetype)
+}
+
+#' Create Components
+#'
+#' Creates a list populated with components and their information from components_list
+#'
+#' @param components_list A list with the lists of components
+#' @return A list named 'components' which contains each component
+create_components <- function(components_list) {
+  components <- list()
+  for (i in 1:length(components_list)) {
+    component <- components_list[[i]]
+    components[[i]] <- list (
+      self = list(component$self),
+      id = list(component$id),
+      name = list(component$name)
+    )
+  }
+  return(components)
+}
+
+# each list is a component, added through create_components above
+component_details <- list(
+  list(self = "https://domain.org/jira/rest/api/2/component/12313938", id = "12313938", name = "x-core"),
+  list(self = "https://domain.org/jira/rest/api/2/component/12313939", id = "12313939", name = "x-spring")
+)
+
+#' Create Creator
+#'
+#' Creates a creator field with information
+#'
+#' @param self_url URL of API endpoint
+#' @param name Name of creator
+#' @param key Key associated with creator
+#' @param displayName Display name of creator
+#' @param active TRUE, indicates if creator's account is active or not
+#' @param timeZone Timezone associated with creator
+#' @return A list named 'creator' that has creator's information
+#' @export
+create_creator <- function(self_url = "https://domain.org/jira/rest/api/2/user?username=user1",
+                           name = "user1",
+                           key = "user1",
+                           displayName = "Fake User1",
+                           active = TRUE,
+                           timeZone = "Etc/UTC") {
+  creator <- list(
+    self = list(self_url),
+    name = list(name),
+    key = list(key),
+    displayName = list(displayName),
+    active = list(active),
+    timeZone = list(timeZone)
+  )
+
+  return(creator)
+}
+
+#' Create Reporter
+#'
+#' Creates a reporter field with information
+#'
+#' @param self_url URL of API endpoint
+#' @param name Name of reporter
+#' @param key Key associated with reporter
+#' @param displayName Display name of reporter
+#' @param active TRUE, indicates if reporter's account is active or not
+#' @param timeZone Timezone associated with reporter
```

Comment:
You have to be consistent in the code style. Please make sure all your function parameters are `in_this_form` instead of `inThisForm`.